### DISCUSSION TITLE
Reamrk- components code block not rendering...

### DISCUSSION THREAD

**Discussion by AbhayaShankar** [remarkjs/remark#1156] [Q&A] [answered]
``` js
 const customRenderers = {
code({ language, value }) {
      return (
        <SyntaxHighlighter language={language} style={atomDark}>
          {value}
        </SyntaxHighlighter>
      );
    },
  };

<ReactMarkdown
        className={classes.markdownContent}
        components={customRenderers}
        children={content}
      />
```

All though every other content is showing up but the code block is empty

What could have gone wrong?

**Comment by ChristianMurphy** [comment]
Welcome @AbhayaShankar!
Sorry you ran into a spot of trouble.

A good starter would be following the example in the `react-markdown` readme https://github.com/remarkjs/react-markdown#use-custom-components-syntax-highlight

````ts
import React from 'react'
import ReactDom from 'react-dom'
import ReactMarkdown from 'react-markdown'
import {Prism as SyntaxHighlighter} from 'react-syntax-highlighter'
import {dark} from 'react-syntax-highlighter/dist/esm/styles/prism'

// Did you know you can use tildes instead of backticks for code in markdown? ✨
const markdown = `Here is some JavaScript code:

~~~js
console.log('It works!')
~~~
`

ReactDom.render(
  <ReactMarkdown
    children={markdown}
    components={{
      code({node, inline, className, children, ...props}) {
        const match = /language-(\w+)/.exec(className || '')
        return !inline && match ? (
          <SyntaxHighlighter
            {...props}
            children={String(children).replace(/\n$/, '')}
            style={dark}
            language={match[1]}
            PreTag="div"
          />
        ) : (
          <code {...props} className={className}>
            {children}
          </code>
        )
      }
    }}
  />,
  document.body
)
````

If the issue persists, consider sharing a reproducible example of what you are seeing.
Using a [codesandbox](https://codesandbox.io/s/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-create-react-app?file=/src/app.tsx) or [stackblitz](https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-create-react-app?file=src%2Fapp.tsx) as a starter.
It's unclear from what you shared, what libraries, what versions of libraries, and what content you are testing with.

**Reply by AbhayaShankar** [reply]
I believe I have gone through the docs and tried multiple approaches to resolve this issue.
When I didn't know how to resolve this then I posted this so that other developers who might have faced this issue could help me out on this

**Reply by ChristianMurphy** [reply]
Again

> If the issue persists, consider sharing a reproducible example of what you are seeing.
> Using a [codesandbox](https://codesandbox.io/s/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-create-react-app?file=/src/app.tsx) or [stackblitz](https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-create-react-app?file=src%2Fapp.tsx) as a starter.
> It's unclear from what you shared, what libraries, what versions of libraries, and what content you are testing with.

You haven't provided enough information for people to be able to offer much advise.
Help the community help you, by providing enough detail for people to offer tangible suggestions.

**Reply by AbhayaShankar** [reply]
Sure I will do that

**Reply by AbhayaShankar** [reply]
Given a markdown file : 
# NEXTJS 
NEXT Js is the React framework for production - it makes building fullstack React apps and sites a breeze  and ships with built-in SSR/SSG.

``` js
export function getStaticPaths() {
  const postFileNames = getPostsFiles();

  const slugs = postFileNames.map((file) => file.replace(/\.md$/, ""));

  return {
    paths: slugs.map((slug) => ({ params: { slug: slug } })),
    fallback: "blocking",
  };
}   
```


I am developing a blog app for which I want to render code block UI into some styled block for which I have used react-syntax-highlighter and react-markdown for Markdown files.

I then read through these files and render the content on UI.


``` js

import React from "react";
import ReactMarkdown from "react-markdown";
import PostHeader from "./PostHeader";
import classes from "./PostContent.module.css";
import Image from "next/image";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { atomDark } from "react-syntax-highlighter/dist/cjs/styles/prism";

const PostContent = ({ post }) => {
  const { title, image, content, slug } = post;

  const imagePath = `/posts/${slug}/${image}`;

  const customRenderers = { 
    code({ language, value }) {
      return (
        <SyntaxHighlighter language={language} style={atomDark}>
          {value}
        </SyntaxHighlighter>
      );
    },
  };

  return (
    <article className={classes.content}>
      <PostHeader title={title} image={imagePath} />
      <ReactMarkdown
        className={classes.markdownContent}
        components={customRenderers}
        children={content}
      />
    </article>
  );
};

```

## THE ERROR THAT I WAS FACING IS: THE CODE BLOCK IS NOT RENDERING EXCEPT REMAINING THINGS ARE RENDERING JUST FINE...


![error2](https://user-images.githubusercontent.com/113185442/236290747-3a6a802d-0f1b-4a4d-bb2a-0d29423b112e.jpg)

![ERROR THAT I AM FACING](https://user-images.githubusercontent.com/113185442/236290295-0d9a2e1c-ff5d-49fc-9b77-d59ce1c0a4f5.jpg)

**Reply by ChristianMurphy** [reply, marked as answer]
That still is missing a fair bit of detail, aim to create a minimal reproducible example https://stackoverflow.com/help/minimal-reproducible-example
and share it using a sandbox like https://next.new

---

In the absence of that.
It sounds like you are trying to do server side rendering with Next js + server components?
If so, you are likely running into an issue in the syntax highlighter itself, such as https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/493, https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/263#issuecomment-787009697, https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/509

Next seems to recommend using `prism` for syntax highlighting markdown code block in SSR mode https://nextjs.org/blog/markdown and https://nextjs-prism.vercel.app/prism
